src/lib.rs

```diff
@@ -98,10 +99,37 @@ impl SharedPool {
     }
 }
 
+/// Options for what the behavior should be in regards to pending scheduled
+/// executions when the pool is dropped.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+pub enum OnPoolDropBehavior {
+    /// Any pending scheduled executions will be run, but periodic actions will
+    /// not be rescheduled once these have completed.
+    ///
+    /// This is the default behavior.
+    CompletePendingScheduled,
+
+    /// Don't run any pending scheduled executions.
+    DiscardPendingScheduled,
+
+    /// Any pending scheduled executions will be run immediately, rather than
+    /// waiting to be run at the time they were originally scheduled to be run
+    /// at.
+    ///
+    /// Note that the order in which the executions will be run is not
+    /// guaranteed. Any pending executions will be popped from the queue of
+    /// pending executions in the expected order but there's no guarantee as to
+    /// what order the executions will run.
+    RunPendingScheduledImmediately,
```

Comment:
Out of curiosity, what's the rationale for this mode? I don't think anything similar exists for e.g. Java's ScheduledExecutor.

Reply:
I originally thought it would be useful in the r2d2 `reap_connections` scenario I mentioned, but it turns out it's not useful there. I'll remove it.